src/content/docs/bot-protection/reference.mdx

```diff
@@ -329,6 +331,32 @@ their "advertising quality" bot.
   <SvelteKitFiltering slot="sveltekit" />
 </SlotByFramework>
 
+## Bot verification
+
+We support verifying the authenticity of requests from identifiable bots by
+comparing the request IP to published bot IP ranges or performing a reverse
+DNS query. To check the verification status, use the `.isSpoofed()` or `.isVerified()`
+methods on the reason object. If verification was performed, one of these will be true
+and the other false; both will be false if verification failed or is unsupported
+for the identified bot. 
+
+:::note
+It's recommended to check verification status only when using an allow configuration.
+For deny configurations, verification is skipped as you'd want to block listed bots
+regardless of legitimacy.
+:::
```

Comment:
I'm not sure it's that's useful and is an extra step we can skip. It might be interesting to show "requests from bogus/spoofed crawlers" in analytics, but we could calculate that out of band.

Reply:
Yeah I agree, I think its best to return a decision to the user as soon as possible; if the bot's ip isn't cached it would mean we're performing a relatively expensive lookup for no apparent value.

Reply:
The problem I see is that "almost behaving bots" will avoid detection by changing their user agent strings and then we won't do IP lookup. We've seen this with AllenAI in that they released "Dolma" to change the user agent string and avoid previous detections—they could have deployed to different IP ranges but I'm guessing they used the same ones.

Reply:
I think it is a fair point, but I think that's a future expansion of the detect bots feature rather than being related to the bot verification product. We can use the same tools that we've built out in this project to perform this kind of detection though. We can open an issue in the main repo to track this.

Reply:
https://github.com/arcjet/arcjet/issues/3666